src/commands/create.rs

```diff
@@ -1,11 +1,57 @@
-use eyre::{Ok, Result};
+use color_eyre::Section;
+use eyre::{Context, Ok, Result};
+use regex::Regex;
 
-use crate::project::settings::get_project_settings;
+use crate::git_client::get_git_client;
 
 pub fn create() -> Result<()> {
-    let trunk = get_project_settings()?.get_trunk()?;
+    let git_client = get_git_client()?;
 
-    println!("Trunk: {}", trunk);
+    let Some(base_branch) = git_client.get_current_branch() else {
+        return Err(
+            eyre::eyre!("Can't create a new stacked commit without a current branch.")
+                .suggestion("Checkout onto a branch and try again."),
+        );
+    };
+    let temp_branch = git_client.create_branch("gi_temp_branch")?;
+
+    let commit_status = git_client.interactive_commit()?;
+    match commit_status {
+        crate::git_client::CommitStatus::Aborted => {
+            git_client.checkout(&base_branch)?;
+            git_client.delete_branch(&temp_branch)?;
+            return Ok(());
+        }
+        crate::git_client::CommitStatus::Success => {}
+    }
+
+    let commit_title = git_client.get_current_commit_title()?;
+    git_client.create_branch(&format_commit_title(commit_title)?)?;
+    git_client.delete_branch(&temp_branch)?;
 
     Ok(())
 }
+
+/// Formats the commit title by replacing non-alphanumeric characters with underscores and prepending the current date.
+///
+/// # Arguments
+///
+/// * `title` - The original commit title.
+///
+/// # Errors
+///
+/// This function can return an error if the regular expression fails to create a regex pattern.
+///
+/// # Examples
+///
+/// `format_commit_title("Hello, World")` returns `Ok("12-31_Hello_World")`
+fn format_commit_title(title: String) -> Result<String> {
+    // Get day and month from the current date
+    let today = chrono::Local::now().format("%m-%d").to_string();
```

Comment:
Is month day format more suggestive than day month or is it just a matter of preference?

Reply:
When I firstly wrote this, it was actually "day-month". Then I thought that it might be nicer that if when listed, branches would keep the chronological order (in the same year span).

Example:

day-month (chaos)

04-05-...
06-07-...
08-12-...
17-07-...
30-05-...

month-day (chronological)

05-04-...
05-30-...
07-06-...
07-17-...
12-08-...